pkg/controller/factory/base_controller.go

```diff
@@ -171,7 +171,14 @@ func (c *baseController) runWorker(queueCtx context.Context) {
 
 // reconcile wraps the sync() call and if operator client is set, it handle the degraded condition if sync() returns an error.
 func (c *baseController) reconcile(ctx context.Context, syncCtx SyncContext) error {
-	err := c.sync(ctx, syncCtx)
+	var err error
```

Comment:
push this to where the go func is started, like kube. That's where the risk starts.  Also, use utilrutnime.HandleCrash to be consistent with literally every other binary we have that honors `OPENSHIFT_ON_PANIC`.  We even wire this in `library-go/pkg/controller/contrrollercmd`